bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -176,6 +194,29 @@ private void updateChannel(String channelId) {
         if (isLinked(channelId)) {
             State state = null;
             SonnenJsonDataDTO data = serviceCommunication.getBatteryData();
+            // String direction = GetPowerMeterDirection(channelId);
+
+            // The sonnen API has two sub-channels, e.g. 4_1 and 4_2, one representing consumption and the
+            // other production. E.g. 4_1.kwh_imported represents the total production since the
+            // battery was installed.
+            SonnenJsonPowerMeterDataDTO[] dataPM = null;
+            if (config.powerMeter) {
+                dataPM = serviceCommunication.getPowerMeterDatas();
+            }
+
+            if (dataPM != null) {
+                switch (channelId) {
+                    case KWHIMPORTEDSTATE:
+                        state = new QuantityType<Power>(dataPM[1].getKwhImported(), Units.WATT);
```

Comment:
This unit doesn't match the definition of the channel type which is `Number:Energy`. If your value is Wh, you could use:
```suggestion
                        state = new QuantityType<Energy>(dataPM[1].getKwhImported(), Units.WATT_HOUR);
```

or if kWh:
```suggestion
                        state = new QuantityType<Energy>(dataPM[1].getKwhImported(), Units.KILOWATT_HOUR);
```

Reply:
I believe wrong version was picked. Since the API name contains "kwh" I would assume it's kWh, not Wh.

Reply:
fixed